packages/babel-parser/src/types.js

```diff
@@ -965,7 +965,8 @@ export type TsTypeParameterDeclaration = TypeParameterDeclarationBase & {
 };
 
 export type TypeParameterBase = NodeBase & {
-  name: string,
+  // TODO(Babel-8): remove string type support
+  name: string | Identifier,
 };
 
 export type TypeParameter = TypeParameterBase & {
```

Comment:
`TypeParameter` also depends on `TypeParameterBase`, this must be changed too, right?